### DISCUSSION TITLE
Recommendations on running with AWS Lambda?

### DISCUSSION THREAD

**Discussion by dternyak** [miguelgrinberg/Flask-SocketIO#1687] [Q&A] [answered]
Hi @miguelgrinberg 

Have you seen Flask-Socketio running in a Lambda environment yet with API Gateway? Do you have any best practices around how you would approach it?

In API Gateway, the connections are "held" outside of the lambda which only fires to process the event. Given this, a message queue to route emits to the correct worker no longer makes sense.

The entire model seems to change a lot.

I'd love to hear your take on how we might approach this, or if you've seen anyone utilize API Gateway with Flask-Socketio or Socketio in general so far.

**Comment by miguelgrinberg** [comment, marked as answer]
I have not done this myself, or seen anyone else do it, sorry. It might be possible with the [WebSocket support in API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api.html), but it needs some research.

**Reply by dternyak** [reply]
Got it, thank you!
